conan/tools/microsoft/__init__.py

```diff
@@ -2,3 +2,17 @@
 from .toolchain import MSBuildToolchain
 from .msbuild import MSBuild
 from .msbuilddeps import MSBuildDeps
+
+
+def msvc_runtime_flag(conanfile):
+    settings = conanfile.settings
+    compiler = settings.get_safe("compiler")
```

Comment:
```suggestion
    compiler = self.settings.get_safe("compiler.base") or settings.get_safe("compiler")
```

Reply:
I'd prefer to condition it to ``intel`` compiler explicitly.

Reply:
up to you :)